navigation/AppNavigation.js

```diff
@@ -1,13 +1,35 @@
+import React from 'react';
 import { createStackNavigator, createAppContainer } from 'react-navigation';
 
 import LibraryContainer from '../app/screens/Library/index';
 import BookDetailContainer from '../app/screens/BookDetail/index';
 
+import ImageHeader from './components/ImageHeader';
+import BackButtonHeader from './components/BackButtonHeader';
+import { styles } from './styles';
+
 const AppNavigator = createStackNavigator({
-  library: { screen: LibraryContainer },
-  bookDetail: { screen: BookDetailContainer }
-}, {
-  initialRouteName: 'library'
-});
+    library: {
+      screen: LibraryContainer,
+      navigationOptions: {
+        title: 'Library'
+      }
+    },
+    bookDetail: {
+      screen: BookDetailContainer,
+      navigationOptions: {
+        title: 'Book Detail'
+      }
+    }
+  },
+  {
+    initialRouteName: 'library',
+    defaultNavigationOptions: {
+      headerBackground: <ImageHeader />,
+      headerTitleStyle: styles.titleStyle,
+      headerBackImage: <BackButtonHeader />
```

Comment:
same with `headerBackImage` you don't really need the jsx. Check https://github.com/react-navigation/react-navigation/blob/2c7187b22aeff1cdec5ca6aeebb40c9c798c0888/examples/NavigationPlayground/js/StackWithCustomHeaderBackImage.js